modules/dnn/src/ocl4dnn/src/ocl4dnn_conv_spatial.cpp

```diff
@@ -1822,13 +1794,25 @@ bool OCL4DNNConvSpatial<Dtype>::loadCachedConfig()
     cv::AutoLock lock(kernelConfigMutex);
     if (!defaultConfigLoaded)
     {
-        const size_t numConfigs = sizeof(default_kernel_config_intel)/sizeof(default_kernel_config_intel[0])/2;
+        std::string prefix = std::string("Intel(R) Corporation_");
+        size_t numConfigs;
+        if (use_half_)
+            numConfigs = sizeof(default_kernel_config_intel_fp16) / sizeof(const char *) / 2;
+        else
```

Comment:
`defaultConfigLoaded` is global variable. So both configs should be initialized at once.

Reply:
@alalek , PR is rebased  on latest 3.4 branch.